modules/kameleonvolume/rendering/renderablekameleonvolume.cpp

```diff
@@ -0,0 +1,375 @@
+    }
+
+    glm::vec3 lowerDomainBound;
+    if (dictionary.getValue("LowerDomainBound", lowerDomainBound)) {
+        _lowerDomainBound = lowerDomainBound;
+    }
+    else {
+        _autoDomainBounds = true;
+    }
+
+    glm::vec3 upperDomainBound;
+    if (dictionary.getValue("UpperDomainBound", upperDomainBound)) {
+        _upperDomainBound = upperDomainBound;
+    }
+    else {
+        _autoDomainBounds = true;
+    }
+
+    glm::vec3 domainScale;
+    if (dictionary.getValue("DomainScale", domainScale)) {
+        _domainScale = domainScale;
+    } else {
+        _domainScale = glm::vec3(1, 1, 1); // Assume meters if nothing else is specified.
+    }
+
+    float lowerValueBound;
+    if (dictionary.getValue("LowerValueBound", lowerValueBound)) {
+        _lowerValueBound = lowerValueBound;
+    }
+    else {
+        _autoValueBounds = true;
+    }
+
+    float upperValueBound;
+    if (dictionary.getValue("UpperValueBound", upperValueBound)) {
+        _upperValueBound = upperValueBound;
+    }
+    else {
+        _autoValueBounds = true;
+    }
+
+    ghoul::Dictionary clipPlanesDictionary;
+    dictionary.getValue("ClipPlanes", clipPlanesDictionary);
+    _clipPlanes = std::make_shared<VolumeClipPlanes>(clipPlanesDictionary);
+    _clipPlanes->setName("clipPlanes");
+
+    bool cache;
+    if (dictionary.getValue("Cache", cache)) {
+        _cache = cache;
+    }
+
+    _gridType.addOption(static_cast<int>(VolumeGridType::Cartesian), "Cartesian grid");
+    _gridType.addOption(static_cast<int>(VolumeGridType::Spherical), "Spherical grid");
+    _gridType.setValue(static_cast<int>(VolumeGridType::Cartesian));
+
+    std::string gridType;
+    if (dictionary.getValue("GridType", gridType)) {
+        if (gridType == "Spherical") {
+            _gridType.setValue(static_cast<int>(VolumeGridType::Spherical));
+        } else {
+            _autoGridType = true;
+        }
+    }
+}
+    
+RenderableKameleonVolume::~RenderableKameleonVolume() {}
+
+bool RenderableKameleonVolume::initialize() {
+    load();
+    
+    _volumeTexture->uploadTexture();
+    _transferFunction->update();
+
+    _raycaster = std::make_unique<KameleonVolumeRaycaster>(_volumeTexture, _transferFunction, _clipPlanes);
+
+    _raycaster->setStepSize(_stepSize);
+    _gridType.onChange([this] {
+        _raycaster->setStepSize(_stepSize);
+    });
+    _raycaster->setGridType(static_cast<VolumeGridType>(_gridType.value()));
+    _gridType.onChange([this] {
+        _raycaster->setGridType(static_cast<VolumeGridType>(_gridType.value()));
+    });
+
+    updateRaycasterModelTransform();
+    _lowerDomainBound.onChange([this] {
+        updateRaycasterModelTransform();
+    });
+    _upperDomainBound.onChange([this] {
+        updateRaycasterModelTransform();
+    });
+
+    _raycaster->initialize();
+
+    OsEng.renderEngine().raycasterManager().attachRaycaster(*_raycaster.get());
+
+    std::function<void(bool)> onChange = [&](bool enabled) {
```

Comment:
better:
`auto onChange =...`
as it enables more optimization (`std::function` is pretty expensive)